packages/strapi-design-system/src/DatePicker/DatePickerCalendar.tsx

```diff
@@ -54,98 +74,328 @@ export const DatePickerCalendarContent = ({
-                    </VisuallyHidden>
-                  </DatePickerTd>
-                );
-              })}
+                    isSelected={selectedDate ? isSameDay(convertUTCDateToCalendarDate(selectedDate), date) : false}
+                  />
+                ) : (
+                  <Cell />
+                ),
+              )}
             </RawTr>
           ))}
         </RawTbody>
       </RawTable>
-    </Box>
+    </Flex>
+  );
+};
+
+const convertUTCDateToCalendarDate = (date: Date): CalendarDate => {
+  const utcDateString = date.toISOString();
+  const zonedDateTime = parseAbsoluteToLocal(utcDateString);
+
+  /**
+   * ZonedDateTime can't have weeks added,
+   * see – https://github.com/adobe/react-spectrum/issues/3667
+   */
+  return toCalendarDate(zonedDateTime);
+};
+
+const makeGetDatesInWeek = (from: CalendarDate, locale: string) => (weekIndex: number) => {
+  let date = from.add({ weeks: weekIndex });
+  let dates: Array<CalendarDate | null> = [];
+
+  date = startOfWeek(date, locale);
+
+  /**
+   * startOfWeek will clamp dates within the calendar system's valid range, which may
+   * start in the middle of a week. In this case, add null placeholders.
+   */
+  let dayOfWeek = getDayOfWeek(date, locale);
+  for (let i = 0; i < dayOfWeek; i++) {
+    dates.push(null);
+  }
+
+  while (dates.length < 7) {
+    dates.push(date);
+    let nextDate = date.add({ days: 1 });
+
+    if (isSameDay(date, nextDate)) {
+      /**
+       * If the next day is the same, we have hit the end of the calendar system.
+       */
+      break;
+    }
+
+    date = nextDate;
+  }
+
+  /**
+   * Add null placeholders if at the end of the calendar system.
+   */
+  while (dates.length < 7) {
+    dates.push(null);
+  }
+
+  return dates;
+};
+
+const ToolbarFlex = styled(Flex)`
+  div[role='combobox'] {
+    border: 1px solid transparent;
+    background: transparent;
+    font-weight: ${(props) => props.theme.fontWeights.bold};
+
+    ${Typography} {
+      color: ${({ theme }) => theme.colors.neutral800};
+    }
+
+    svg {
+      > g,
+      path {
+        fill: ${({ theme }) => theme.colors.neutral500};
+      }
+    }
+
+    &:hover {
+      background-color: ${({ theme }) => theme.colors.neutral100};
+    }
+  }
+`;
+
+/* -------------------------------------------------------------------------------------------------
+ * HeaderCell
+ * -----------------------------------------------------------------------------------------------*/
+
+interface HeaderCellProps extends Omit<RawThProps, 'children'> {
+  children: string;
+}
+
+const HeaderCell = ({ children, ...props }: HeaderCellProps) => {
+  return (
+    <Th {...props}>
+      <Flex height={`${24 / 16}rem`} width={`${32 / 16}rem`} justifyContent="center">
+        <Typography variant="pi" fontWeight="bold" color="neutral800" aria-hidden>
+          {children.slice(0, 2)}
+        </Typography>
+        <VisuallyHidden>{children}</VisuallyHidden>
```

Comment:
Could this just be an aria-label instead to make the HTML structure a bit smaller? This would also avoid `aria-hidden` on Typography.